### DISCUSSION TITLE
Fragen zur Sprache

### DISCUSSION THREAD

**Discussion by Thepigcat76** [BenediktLamberty/EigeneProgrammiersprache#4] [General]
Ich habe dich durch den KI wettbewerb gefunden und habe mir dein Profil angesehen. Dabei ist mir dieses Repository ganz besonders aufgefallen, da ich auch an einer Programmiersprache arbeite.

1. Ist die Sprache interpretiert oder kompiliert?
2. Wieso macht ihr sie?
3. Wofür nutzt ihr assembly?
4. Wieso habt ihr euch für python entschieden?

Ich bin einfach nur neugierig und möchte ein bisschen 'Erfahrung' sammeln. Ist schwierig im Internet festzustellen, deshalb dachte ich mir, mache ich diese Anmerkung :)

**Comment by BenediktLamberty** [comment]
Die Sprache wird mit dem Python-Code in MIPS-Assembly kompiliert. Den Assemblycode führe ich dann in einem MIPS-Simulator Mars aus. 
Ich benutze Python, weil ich mit dieser Sprache am meisten vertraut bin. Eigendlich hatte ich keine großen Hintergedanken. 
Das mit dem Assembly habe ich auf einem Workshop des Bundeswettbewerbs Informatik kennengelernt. Da haben wir auch so einen Compiler geschrieben. Eigendlich ist dieses Projekt hier nur eine Fortsetzung davon, weil ich an dem Thema interessiert war. Einfach ein kleines Freizeitprojekt ohne großen Nutzen ^^
 
Wie du siehst, habe ich  in der letzten Zeit nicht wirklich daran weitergearbeitet. Allerdings will ich eigendlich noch weitere Features hinzufügen. 

Ich hätte auch noch ein paar Fragen zurück ;)

- Warst du schon mal beim BWKI (Ich denke mal, das ist der KI-Wettbewerb, von dem du sprichst) dabei und wie wars so? Ich habe noch keine Ahnung von KI und will mir das in der nächsten Zeit mal aneignen. Mein Informatiklehrer hat mich da eigendlich etwas Hals über Kopf reingeworfen ^^

- Worum handelt es sich bei deiner Sprache (ist das der coro-interpreter?) ?

**Reply by Thepigcat76** [reply]
1. Nein, ich war noch nie beim BWKI und bin mir nicht mehr so sicher, ob ich mit machen möchte, da ich vom letzten Projekt ziemlich geburntout bin.

2. Ja es it der coro-interpreter, den ich für meine App coda-rover, eine moderne Version von Robot Karol schreibe.

Ach ja, wie weit bist du mit deiner Sprache?

Und noch eine etwas off-topic Frage, ich habe gesehen, dass ihr ein game macht und wollte wissen worum es geht und welche engine/frameworks ihr hernehmt.

Liebe Grüße, der andere Benedikt.

**Reply by BenediktLamberty** [reply]
Ahh ein Benedikt. I like it^^

Der Compiler schafft es schon, Funktionen, Structs, Schleifen, Fallunterscheidungen und Expressions in einen AST zu übersetzten. Der Compiler kann allerdings bis jetzt nur die Kontrollstrukturen und  Expressions in MIPS kompilieren. In MIPS werden dann die Expressions auf dem Stack evaluiert und Kontrollstrukturen mit Branches umgesetzt. Der nächste Schritt, den ich versuchen werde umzusetzen ist das aufrufen von Funktionen und der Aufbau von Stackframes. 

Ich bin auf meiner Schule Mitbegründer einer Game-Development-AG, in der wir mit Unity ein Spiel programmieren. Das Spiel soll ein 2D Platformer / Puzzle-Game werden, in dem eine Seele aus der Hölle entkommen muss und dabei in verschiedene Körper eindringen und diese in Besitz nehmen kann.

**Reply by Thepigcat76** [reply]
Das hört sich ziemlich beeindruckend an! Vorallem, dass du die Sprache ohne LLVM machst. Viel Erfolg dir noch! Muss jetzt noch mein eigenes Game fertig machen. Mal schauen, vielleicht kann ich ja nochmal was contributen oder wir können mal ein Projekt gemeinsam machen.

**Reply by BenediktLamberty** [reply]
Dann auf jeden Fall viel Spaß bei deinem Game! Wir könne gerne mal in Zukunft was zusammen machen, allerdings habe ich zur Zeit mit meinen Projekt, Der AG und dem BWKI genug zu tun
